operator-common/src/main/java/io/strimzi/operator/cluster/model/Ca.java

```diff
@@ -1187,4 +1252,33 @@ protected boolean hasCaCertGenerationChanged(Secret secret) {
         }
         return false;
     }
+
+    /**
+     * Generates hash stub of the certificate which is used to track when the certificate changes and rolling update needs to be triggered.
+     *
+     * @param certSecret    Secrets with the certificate
+     * @param key           Key under which the certificate is stored in the Secret
+     *
+     * @return              Hash stub of the certificate
+     */
+    public static String getCertificateThumbprint(Secret certSecret, String key) {
+        try {
+            X509Certificate cert = Ca.cert(certSecret, key);
+            return Util.hashStub(cert.getEncoded());
```

Comment:
IIRC (but might be wrong) the OpenSSL fingerprint is the whole SHA1 of the DER, not just the prefix. If that's correct let's avoid using a truncation here and return the whole thing.

Reply:
I'm fine with that.

Note that this is already in use for the `ANNO_STRIMZI_CUSTOM_LISTENER_CERT_THUMBPRINTS` annotation, but I don't think this would a very far reaching breaking change. Users will end up with bigger value in that annotation after upgrade , so pods will be rolled even though the certs are the same, but that's just during the upgrade.

Reply:
@tombentley after some discussion with @scholzj , it seems using the short 8-characters SHA1 hash is a pattern used in multiple places in the project, and here it might have the added benefit of easier diffing when changed, is there a particular benefit that you see that would make us not use `hashStub` here and just generate the full SHA1 hash?

Reply:
> is there a particular benefit that you see that would make us not use `hashStub` here

Only to the extent that a `SHA1(DER)` is a thing which is spat out by other tools in the SSL world, so there is some very minor utility being consistent with that. Using a truncated version of that still works, but it's no longer "the certificate fingerprint". I can live with it either way.

Reply:
changed to the full sha1.